bootstore/src/schemes/v0/peer.rs

```diff
@@ -0,0 +1,802 @@
+                    handle: accepted_handle.handle,
+                    tx: accepted_handle.tx,
+                    addr,
+                    unique_id: accepted_handle.unique_id,
+                };
+                self.connections.insert(peer_id.clone(), handle);
+                if let Err(e) =
+                    self.fsm.on_connected(Instant::now().into(), peer_id)
+                {
+                    // This can only be a failure to init the rack, so we
+                    // log it as an error and not a warning. It is unrecoverable
+                    // without a rack reset.
+                    error!(self.log, "Error on connection: {e}");
+                } else {
+                    self.deliver_envelopes().await;
+                }
+            }
+            ConnToMainMsgInner::ConnectedClient { addr, peer_id } => {
+                let handle =
+                    self.negotiating_connections.remove(&addr).unwrap();
+
+                // Ignore the stale message if the unique_id doesn't match what
+                // we have stored.
+                if unique_id != handle.unique_id {
+                    return;
+                }
+
+                self.connections.insert(peer_id.clone(), handle);
+                if let Err(e) =
+                    self.fsm.on_connected(Instant::now().into(), peer_id)
+                {
+                    // This can only be a failure to init the rack, so we
+                    // log it as an error and not a warning. It is unrecoverable
+                    // without a rack reset. It's not an invariant violation
+                    // though, so we don't panic.
+                    error!(self.log, "Error on connection: {e}");
+                } else {
+                    self.deliver_envelopes().await;
+                }
+            }
+            ConnToMainMsgInner::Disconnected { peer_id } => {
+                // Ignore the stale message if the unique_id doesn't match what
+                // we have stored.
+                if let Some(handle) = self.connections.get(&peer_id) {
+                    if unique_id != handle.unique_id {
+                        return;
+                    }
+                }
+                warn!(self.log, "peer disconnected {peer_id}");
+                self.connections.remove(&peer_id);
+                self.fsm.on_disconnected(&peer_id);
+            }
+            ConnToMainMsgInner::Received { from, msg } => {
+                match self.fsm.handle_msg(Instant::now().into(), from, msg) {
+                    Ok(None) => self.deliver_envelopes().await,
+                    Ok(Some(api_output)) => {
+                        self.handle_api_output(api_output).await
+                    }
+                    Err(err) => self.handle_api_error(err).await,
+                }
+            }
+            ConnToMainMsgInner::FailedServerHandshake {
+                addr: SocketAddrV6,
+            } => {}
```

Comment:
In `peer_networking.rs`, it mentions returning this error so the main task can handle "cleaning up any metadata".  Is there nothing to clean up?

Reply:
Good catch! fixed in dcd5dac0d